### PR TITLE
NO-ISSUE: Publish with Yalc after build in dev environment

### PR DESCRIPTION
**Pull request by jtomasek**
[Yalc](https://github.com/wclr/yalc) - a tool to locally publish a package so it can be easily used and tested by a consuming application

Manual process (for details see the [docs](https://github.com/wclr/yalc)):
1. Publish the package locally with `yalc publish` (in assisted-ui-lib directory)
2. Add the package to consuming app with `yalc link openshift-assisted-ui-lib` (in consuming app directory e.g. assisted-ui)
3. When a assisted-ui-lib is built, run `yalc push` to publish the package locally and update it in all consuming apps that have it linked

This pull request aims to automate the third step so every time an assisted-ui-lib is built it is automatically updated in all consuming apps that have it linked.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,7 +84,8 @@
     "js-yaml": "^4.1.0",
     "prism-react-renderer": "^1.1.1",
     "react-measure": "^2.5.2",
-    "unique-names-generator": "^4.2.0"
+    "unique-names-generator": "^4.2.0",
+    "yalc": "^1.0.0-pre.53"
```
Comment (jkilzi) [inline package.json:88]: Shouldn't it be a devDependency?

File: scripts/esbuild/dev.js [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
 esbuild
   .build({
     ...commonConfig,
     watch: {
       onRebuild(error, result) {
         if (error) console.error('Watch build failed:', error);
-        else console.log('Watch build succeeded:', result);
+        else {
+          console.log('Watch build succeeded:', result);
+          publishWithYalc();
```
Comment (jkilzi) [inline scripts/esbuild/dev.js:17]: Maybe we could log the publish step has finished after line 17

File: scripts/esbuild/dev.js [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+        }
       },
     },
   })
-  .then((result) => console.log('Build succeeded and started watching', result));
+  .then((result) => {
+    console.log('Build succeeded and started watching', result);
+    publishWithYalc();
```
Comment (jkilzi) [inline scripts/esbuild/dev.js:24]: Same log here...
```suggestion
    publishWithYalc();
    console.log("Published!")
```

File: scripts/esbuild/dev.js (hunk 2 above)
Comment (jkilzi) [reply scripts/esbuild/dev.js:17]: btw, what if the call to  `publishWithYalc` fails? That would be a very bad DevX... maybe we should at least wrap the call in a catch clause, wdyt?

**Review by jkilzi — CHANGES_REQUESTED** [review]
Some questions...

File: package.json (hunk 1 above)
Comment (jtomasek) [reply package.json:88]: Yes

File: scripts/esbuild/dev.js (hunk 2 above)
Comment (jtomasek) [reply scripts/esbuild/dev.js:17]: Considering I have the package linked in 2 projects (uhc-portal and stolostron/console), the output of a build is following:
```
Build succeeded and started watching { errors: [], warnings: [], stop: [Function: stop] }
openshift-assisted-ui-lib@2.4.0 published in store.
Pushing openshift-assisted-ui-lib@2.4.0 in /root/assisted-ui
Package openshift-assisted-ui-lib@2.4.0 linked ==> /root/assisted-ui/node_modules/openshift-assisted-ui-lib
Pushing openshift-assisted-ui-lib@2.4.0 in /root/open-cluster-management-console/frontend
Package openshift-assisted-ui-lib@2.4.0 linked ==> /root/open-cluster-management-console/frontend/node_modules/openshift-assisted-ui-lib
```

Isn't that sufficient?

[the remark this answers is not in the document]
Comment (jtomasek) [reply scripts/esbuild/dev.js:17]: Done

File: scripts/esbuild/dev.js [hunk 4]
```diff
@@ -1,15 +1,36 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
 const esbuild = require('esbuild');
 const commonConfig = require('./common-config');
+const { publishPackage } = require('yalc');
+const { join } = require('path');
+
+function publishWithYalc() {
+  try {
+    publishPackage({
+      workingDir: join(process.cwd() || ''),
```
Comment (jkilzi) [inline scripts/esbuild/dev.js:10]:
```suggestion
      workingDir: join(process.cwd()),
```

**Review by jkilzi — DISMISSED** [review]

**Review by jkilzi — APPROVED** [review]

**Pull request merged**